receiver/awscontainerinsightreceiver/internal/k8sapiserver/metric_scrape.go

```diff
@@ -0,0 +1,171 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sapiserver
+
+import (
+	"context"
+	"time"
+
+	gokitLog "github.com/go-kit/log"
+	httpconfig "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/prometheus/prometheus/scrape"
+	"github.com/prometheus/prometheus/storage"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/logging"
+)
+
+const (
+	collectionEndpoint = "/metrics"
+	bearerTokenPath    = "/var/run/secrets/kubernetes.io/serviceaccount/token" // #nosec
+	collectionInterval = 15 * time.Second                                      // TODO:
+	collectionTimeout  = 7 * time.Second                                       // TODO
+)
+
+type MetricScrape struct {
+	scrapeManager    *scrape.Manager
+	discoveryManager *discovery.Manager
+	logger           *zap.Logger
+	gokitlogger      gokitLog.Logger
+	store            *storage.Appendable
+	config           *config.Config
+	cancel           context.CancelFunc
+	context          context.Context
+}
+
+func NewMetricScrape(host string, logger *zap.Logger, store *storage.Appendable) (*MetricScrape, error) {
+	// TODO
+	/* 2023-05-16T18:00:47.956Z        debug   scrape/scrape.go:1351   Scrape failed   {"kind": "receiver", "name": "awscontainerinsightreceiver", "data_type": "metrics", "scrape_pool": "k8sapiserver", "target": "https://ip-192-168-71-146.us-east-2.compute.internal:443/metrics", "error": "Get \"https://ip-192-168-71-146.us-east-2.compute.internal:443/metrics\": unable to read authorization credentials file /var/Run/secrets/kubernetes.io/serviceaccount/token: open /var/Run/secrets/kubernetes.io/serviceaccount/token: no such file or directory"}
+	 */
+
+	config := &config.Config{
+		ScrapeConfigs: []*config.ScrapeConfig{
+			{
+				JobName: "k8sapiserver",
+
+				ScrapeInterval: model.Duration(collectionInterval),
+				ScrapeTimeout:  model.Duration(collectionTimeout),
+
+				HonorLabels:     true,
+				HonorTimestamps: true,
+
+				MetricsPath: collectionEndpoint,
+				Scheme:      "https",
+
+				HTTPClientConfig: httpconfig.HTTPClientConfig{
+					TLSConfig: httpconfig.TLSConfig{
+						InsecureSkipVerify: true,
+					},
```

Comment:
Is this required for production or is this just for getting started?

Reply:
Unless we can provide the CA File and Cert file, this is what we have to do.  We do the same in CWA:

https://github.com/aws/amazon-cloudwatch-agent/blob/a46c6451609305fbbc049b7e12f0616f1170f832/internal/k8sCommon/kubeletutil/kubeletclient.go#L40

Maybe we can find them as secrets? Not sure